torchinductor/utils.py

```diff
@@ -165,3 +165,50 @@ def wrapper(self):
         return getattr(self, key)
 
     return wrapper
+
+
+def sympy_str(expr: sympy.Expr):
+    """
+    Normal sympy str is very slow, this is a lot faster.  The result are
+    somewhat worse, as it doesn't do as much simplification.  So don't
+    use this for final codegen.
+    """
+    if isinstance(expr, sympy.Symbol):
+        return expr.name
+    if isinstance(expr, sympy.Add):
+        return " + ".join(map(sympy_str, expr.args))
```

Comment:
You may be able to get the equivalent of this if you use `sympy.sstr(expr, order='none')`

Reply:
Which should make the rest of this function unnecessary as well.

Reply:
I tried replacing this with `return sympy.sstr(expr, order='none')`, overall ~39% slower:
```
         18643652 function calls (17885889 primitive calls) in 9.723 seconds

   Ordered by: cumulative time
   List reduced from 1726 to 100 due to restriction <100>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000    9.748    9.748 compile_fx.py:42(compile_fx_inner)
      6/3    0.000    0.000    9.716    3.239 utils.py:70(time_wrapper)
     1769    0.050    0.000    6.351    0.004 interpreter.py:95(run)
        1    0.000    0.000    6.256    6.256 graph.py:332(compile_to_fn)
        1    0.000    0.000    6.256    6.256 graph.py:315(compile_to_module)
        1    0.000    0.000    6.157    6.157 graph.py:307(codegen)
    35434    0.104    0.000    5.145    0.000 interpreter.py:144(run_node)
        1    0.004    0.004    4.560    4.560 scheduler.py:502(__init__)
     2449    0.022    0.000    3.921    0.002 dependencies.py:240(extract_read_writes)
      484    0.006    0.000    3.567    0.007 scheduler.py:220(__init__)
     1768    0.016    0.000    3.324    0.002 ir.py:3288(__call__)
     1768    0.007    0.000    3.214    0.002 ir.py:3388(__call__)
        1    0.000    0.000    3.172    3.172 graph.py:136(run)
     3102    0.014    0.000    3.160    0.001 graph.py:294(run_node)
38586/24454    0.027    0.000    2.911    0.000 printer.py:290(doprint)
91842/24454    0.351    0.000    2.892    0.000 printer.py:294(_print)
27675/27674    0.024    0.000    2.818    0.000 printer.py:371(__call__)
27675/27674    0.026    0.000    2.794    0.000 str.py:982(sstr)
7411/7147    0.047    0.000    2.542    0.000 str.py:50(_print_Add)
     8877    0.012    0.000    2.473    0.000 utils.py:170(sympy_str)
    20973    0.028    0.000    2.465    0.000 interpreter.py:236(call_method)
     6879    0.020    0.000    2.338    0.000 dependencies.py:186(load)
      484    0.016    0.000    2.304    0.005 ir.py:1849(simplify_and_reorder)
     4013    0.003    0.000    2.143    0.001 utils.py:161(wrapper)
     1198    0.009    0.000    2.091    0.002 ir.py:1778(get_read_writes)
      968    0.003    0.000    1.995    0.002 ir.py:3225(__init__)
      968    0.015    0.000    1.991    0.002 ir.py:3310(__init__)
8652/1216    0.017    0.000    1.958    0.002 lowering.py:272(inner_fn)
8652/1216    0.012    0.000    1.934    0.002 lowering.py:277(<listcomp>)
      910    0.003    0.000    1.737    0.002 ir.py:341(store_output)
     7411    0.015    0.000    1.483    0.000 printer.py:340(_as_ordered_terms)
7414/7413    0.052    0.000    1.467    0.000 expr.py:1113(as_ordered_terms)
        1    0.003    0.003    1.386    1.386 scheduler.py:1012(codegen)
     2711    0.007    0.000    1.357    0.001 graph.py:226(call_function)
7825/2711    0.031    0.000    1.348    0.000 lowering.py:154(wrapped)
      303    0.003    0.000    1.329    0.004 triton.py:1044(codegen_nodes)
      303    0.006    0.000    1.313    0.004 triton.py:1122(codegen_node_schedule)
     2935    0.009    0.000    1.265    0.000 ir.py:1702(loader)
7414/7413    0.144    0.000    1.263    0.000 expr.py:1167(as_terms)
    16048    0.038    0.000    1.244    0.000 proxy.py:52(create_proxy)
      768    0.003    0.000    1.027    0.001 ir.py:398(store_reduction)
     2755    0.009    0.000    1.024    0.000 ir.py:1288(loader)
     4272    0.006    0.000    1.017    0.000 sizevars.py:488(load)
      582    0.003    0.000    0.983    0.002 ir.py:3199(num_reads)
```